code/WorkInProgress/nadir_antenna.dm

```diff
@@ -56,6 +58,31 @@ var/global/obj/machinery/communications_dish/transception/transception_array
 				src.failsafe_active = FALSE
 			else
 				src.attempt_restart()
+		else if(length(direct_queue) && primed && !src.is_transceiving)
+			var/obj/queued_item = pick(direct_queue)
+			for_by_tcl(transc_pad, /obj/machinery/transception_pad)
+				if(transc_pad.is_transceiving)
+					continue
+				var/datum/powernet/pad_powernet = transc_pad.get_direct_powernet()
+				if(!pad_powernet)
```

Comment:
should this also check if the pad is powered, not just that is has a pnet?

Reply:
Nope - there are two reasons that the powernet is checked. One is to make sure it's cabled at all, and skip it if it isn't; the other is to capture the powernet's number, to then pass to can_transceive (which also takes into account pad power and other factors).